### PR TITLE
[new release] qmp (0.20.0)

### PR DESCRIPTION
**Pull request by psafont**
OCaml implementation of a Qemu Message Protocol (QMP) client

- Project page: <a href="https://github.com/xapi-project/ocaml-qmp">https://github.com/xapi-project/ocaml-qmp</a>

##### CHANGES:

* Update cmdliner usage to latest version

### CODE REVIEW

File: packages/qmp/qmp.0.20.0/opam [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+build: [
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "runtest" "-p" name "-j" jobs] {with-test}
+]
+depends: [
+  "ocaml"
+  "base-unix"
+  "dune" {>= "1.4.0"}
+  "yojson" {>= "1.6.0"}
+  "cmdliner"
```
Comment (mseri) [inline packages/qmp/qmp.0.20.0/opam:20]:
```suggestion
  "cmdliner" {>= "1.1.0"}
```
The `Cmd` module was introduced in this version:
```
#=== ERROR while compiling qmp.0.20.0 =========================================#
# context              2.2.0~alpha3~dev | linux/x86_64 | ocaml-base-compiler.4.02.3 | pinned(https://github.com/xapi-project/ocaml-qmp/releases/download/v0.20.0/qmp-0.20.0.tbz)
# path                 ~/.opam/4.02/.opam-switch/build/qmp.0.20.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p qmp -j 255
# exit-code            1
# env-file             ~/.opam/log/qmp-9-6c2e09.env
# output-file          ~/.opam/log/qmp-9-6c2e09.out
### output ###
# (cd _build/default && /home/opam/.opam/4.02/bin/ocamlc.opt -w -40 -g -bin-annot -I cli/.cli.eobjs/byte -I /home/opam/.opam/4.02/lib/cmdliner -I /home/opam/.opam/4.02/lib/seq -I /home/opam/.opam/4.02/lib/yojson -I lib/.libqmp.objs/byte -no-alias-deps -o cli/.cli.eobjs/byte/cli.cmo -c -impl cli/cli.ml)
# File "cli/cli.ml", line 52, characters 2-7:
# Error: Unbound module Cmd
```

File: packages/qmp/qmp.0.20.0/opam [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+homepage: "https://github.com/xapi-project/ocaml-qmp"
+bug-reports: "https://github.com/xapi-project/ocaml-qmp/issues"
+license: "LGPL-2.1-only WITH OCaml-LGPL-linking-exception"
+tags: [
+  "org:mirage"
+  "org:xapi-project"
+]
+build: [
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "runtest" "-p" name "-j" jobs] {with-test}
```
Comment (kit-ty-kate) [inline packages/qmp/qmp.0.20.0/opam:13]: Should only the tests be marked as unavailable on 32bit architecture or the package as a whole?
```suggestion
  ["dune" "runtest" "-p" name "-j" jobs] {with-test & arch != "arm32" & arch != "x86_32"}
```
```
#=== ERROR while compiling qmp.0.20.0 =========================================#
# context              2.2.0~alpha3~dev | linux/x86_32 | ocaml-base-compiler.4.14.1 | pinned(https://github.com/xapi-project/ocaml-qmp/releases/download/v0.20.0/qmp-0.20.0.tbz)
# path                 ~/.opam/4.14/.opam-switch/build/qmp.0.20.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune runtest -p qmp -j 127
# exit-code            1
# env-file             ~/.opam/log/qmp-7-3a4a19.env
# output-file          ~/.opam/log/qmp-7-3a4a19.out
### output ###
# (cd _build/default && /home/opam/.opam/4.14/bin/ocamlc.opt -w -40 -g -bin-annot -I lib_test/.messages.eobjs/byte -I /home/opam/.opam/4.14/lib/ounit2 -I /home/opam/.opam/4.14/lib/ounit2/advanced -I /home/opam/.opam/4.14/lib/seq -I /home/opam/.opam/4.14/lib/stdlib-shims -I /home/opam/.opam/4.14/lib/yojson -I lib/.libqmp.objs/byte -no-alias-deps -o lib_test/.messages.eobjs/byte/messages.cmo -c -impl lib_test/messages.ml)
# File "lib_test/messages.ml", line 24, characters 56-66:
# 24 |   "event-powerdown.json",          Event { timestamp = (1258551470, 802380); event = "POWERDOWN"; data = None };
#                                                              ^^^^^^^^^^
# Error: Integer literal exceeds the range of representable integers of type int
```

Comment (psafont) [reply packages/qmp/qmp.0.20.0/opam:13]: Seems like the package as a whole, this looks like a deserialization issue

**Review by kit-ty-kate — APPROVED** [review]

**Comment by kit-ty-kate** [comment]
Thanks

**Pull request merged**